docs/ingestion/concurrent-append-replace.md

```diff
@@ -0,0 +1,145 @@
+
+### Add the task lock through the API
+
+Add the following JSON snippet to your supervisor or ingestion spec if you're using the API:
+
+```json
+"context": {
+   "useConcurrentLocks": true
+}   
+```
+ 
+
+## Task lock types
+
+We recommend that you use the `useConcurrentLocks` context parameter so that Druid automatically determines the task lock types for you. If, for some reason, you need to manually set the task lock types explicitly, you can read more about them in this section.
+
+<details><summary>Click here to read more about the lock types.</summary>
+
+When setting task lock types manually, you need to ensure the following: 
+
+- The append task (with `appendToExisting` set to `true`) has `taskLockType` set to `APPEND` in the task context.
+- The replace task (with `appendToExisting` set to `false`) has `taskLockType` set to `REPLACE` in the task context.
+- The segment granularity of the append task is equal to or finer than the segment granularity of the replace task.
+
+Additionally, keep the following in mind:
+
+- Concurrent append and replace fails if the task with `APPEND` lock uses a coarser segment granularity than the task with the `REPLACE` lock. For example, if the `APPEND` task uses a segment granularity of YEAR and the `REPLACE` task uses a segment granularity of MONTH, you should not use concurrent append and replace.
+
+-  Only a single task can hold a `REPLACE` lock on a given interval of a datasource.
+  
+- Multiple tasks can hold `APPEND` locks on a given interval of a datasource and append data to that interval simultaneously.
+
+#### Add a task lock type to your ingestion job
+
+Next, you need to configure the task lock type for your ingestion job: 
+
+- For streaming jobs, the context parameter goes in your supervisor spec, and the lock type is always `APPEND`
+- For legacy JSON-based batch ingestion, the context parameter goes in your ingestion spec, and the lock type can be either `APPEND` or `REPLACE`. 
+ 
+You can provide the context parameter through the API like any other parameter for ingestion job or through the UI.
+
+##### Add a task lock using the Druid console
+
+As part of the  **Load data** wizard for classic batch (JSON-based ingestion) and streaming ingestion, you can configure the task lock type for the ingestion during the **Publish** step:
+
+- If you set **Append to existing** to **True**, you can then set **Allow concurrent append tasks (experimental)** to **True**.
+- If you set **Append to existing** to **False**, you can then set **Allow concurrent replace tasks (experimental)** to **True**.
+
+##### Add the task lock type through the API
+
+Add the following JSON snippet to your supervisor or ingestion spec if you're using the API:
+
+```json
+"context": {
+   "taskLockType": LOCK_TYPE
+}   
+```
+ 
+The `LOCK_TYPE` depends on what you're trying to accomplish.
```

Comment:
This content provides some intro to what APPEND and REPLACE task locks are for and would make more sense towards the start of the `<details>` section